### PR TITLE
NAS-111196 / 21.10 / Add returns to ACL template methods

### PR DESCRIPTION
**Pull request by anodos325**

### CODE REVIEW

**Review by rick-mesta — APPROVED** [review]

File: src/middlewared/middlewared/plugins/filesystem_/acl_template.py [hunk 1]
```diff
@@ -54,10 +61,14 @@ async def validate_acl(self, data, schema, verrors):
         "acltemplate_create",
         Str("name", required=True),
         Str("acltype", required=True, enum=["NFS4", "POSIX1E"]),
-        List("acl", items=[Ref("posix1e_ace"), Ref("nfs4_ace")], required=True),
+        OROperator(Ref('nfs4_acl'), Ref('posix1e_acl'), name='acl', requried=True),
         register=True
     ))
-    async def create(self, data):
+    @returns(Ref('acltemplate_create'))
```
Comment (sonicaj) [inline src/middlewared/middlewared/plugins/filesystem_/acl_template.py:67]: The returned object would mirror `acltemplate_entry` and not `acltemplate_create` ? If i am accurate, we don't need to add an explicit `returns` as for CRUD methods, middleware will automatically do that for us based on `ENTRY` specification

File: src/middlewared/middlewared/plugins/filesystem_/acl_template.py [hunk 2]
```diff
@@ -83,8 +95,12 @@ async def create(self, data):
             ('attr', {'update': True})
         )
     )
+    @returns(Ref('acltemplate_create'))
```
Comment (sonicaj) [inline src/middlewared/middlewared/plugins/filesystem_/acl_template.py:98]: Same as above

File: src/middlewared/middlewared/plugins/filesystem_/acl_template.py [hunk 3]
```diff
@@ -116,6 +132,7 @@ async def do_update(self, id, data):
         return await self.get_instance(id)
 
     @accepts(Int('id'))
+    @returns()
```
Comment (sonicaj) [inline src/middlewared/middlewared/plugins/filesystem_/acl_template.py:135]: This is not required and middleware will automatically add this for delete call in CRUD services

**Review by sonicaj — APPROVED** [review]
Thanks!

**Pull request merged**